### PR TITLE
fix Aqara L1-350 (ZNXDD01LM) support

### PR DESCRIPTION
**Pull request by nklebedev**
Fix Aqara L1-350 ceiling light support. Current version has broken support of power on behaviour and color temperature restore at startup (report unsupported attributes).

 `aqaraOpple` cluster is added to support additional attributes. After this, 'device temperature' and 'power outage count' seems to have the correct values. 'power_outage_memory' is also operated via `aqaraOpple` cluster.

The following data was observed from `aqaraOpple` cluster:

```
{
    "3": 33,          <- device temperature
    "5": 106,         <- outage count
    "9": 256,         <- ?
    "10": 0,          <- ?
    "12": 10,         <- ?
    "13": 28,         <- ?
    "17": 1,          <- ?
    "100": 0,         <- current state
    "101": 254,       <- brightness
    "102": 153,       <- color temperature
    "103": 425986700, <- = 0x19640A8C 0x1964 - 6500, 0x0A8C - 2700, min and max color temp
    "154": 0,         <- ?
}
```

`{"1299":6500,"1300":2700}`

`{"238":0}`

### CODE REVIEW

File: src/lib/xiaomi.ts [hunk 1]
```diff
@@ -339,11 +351,17 @@ export const numericAttributes2Payload = async (msg: Fz.Message, meta: Fz.Meta,
...
                 }
+            } else if (['ZNXDD01LM'].includes(model.model)) {
+                payload.color_temp = value;
             }
             break;
         case '103':
             if (['RTCZCGQ11LM'].includes(model.model)) {
                 payload.monitoring_mode = getFromLookup(value, {0: 'undirected', 1: 'left_right'});
+            } else if (['ZNXDD01LM'].includes(model.model)) {
+                assertNumber(value);
```
Comment (Koenkk) [inline src/lib/xiaomi.ts:362]: If we don't do anything with it, why do an assertion here? It can only give troubles.

File: src/devices/xiaomi.ts [hunk 2]
```diff
@@ -814,8 +814,16 @@ const definitions: Definition[] = [
         model: 'ZNXDD01LM',
         vendor: 'Xiaomi',
         description: 'Aqara ceiling light L1-350',
-        extend: xiaomiExtend.light_onoff_brightness_colortemp({disableEffect: true, colorTempRange: [153, 370]}),
```
Comment (Koenkk) [inline src/devices/xiaomi.ts:817]: You can keep using extend, use this instead: `extend.light_onoff_brightness_colortemp_color({fromZigbee: [fz.aqara_opple]})`

File: src/devices/xiaomi.ts [hunk 3]
```diff
@@ -814,8 +814,16 @@ const definitions: Definition[] = [
...
+        toZigbee: xiaomiExtend.light_onoff_brightness_colortemp({
+            disableEffect: true, disablePowerOnBehavior: true, disableColorTempStartup: true,
+        }).toZigbee.concat([tz.xiaomi_switch_power_outage_memory]),
+        fromZigbee: xiaomiExtend.light_onoff_brightness_colortemp({
+            disableEffect: true, disablePowerOnBehavior: true, disableColorTempStartup: true,
+        }).fromZigbee.concat([fz.aqara_opple]),
+        exposes: xiaomiExtend.light_onoff_brightness_colortemp({
+            disableEffect: true, disablePowerOnBehavior: true, disableColorTempStartup: true,
+            colorTempRange: [153, 370],
+        }).exposes.concat([e.power_outage_memory(), e.device_temperature(), e.power_outage_count()]),
```
Comment (Koenkk) [inline src/devices/xiaomi.ts:826]: I know you can't test it, but I expect that all the other Xiaomi lights work the same, there I suggest to modify the `xiaomiExtend` instead

File: src/lib/xiaomi.ts (hunk 1 above)
Comment (nklebedev) [reply src/lib/xiaomi.ts]: fixed

File: src/devices/xiaomi.ts (hunk 3 above)
Comment (nklebedev) [reply src/devices/xiaomi.ts]: ok, made an extension of `light_onoff_brightness_colortemp` options, modified the `light_onoff_brightness_colortemp` itself and applied it to all other xiaomi devices

Comment (nklebedev) [reply src/devices/xiaomi.ts]: Oh, I forget to add other devices to aqaraOpple converter. Will push a new commit soon...

**Comment by Koenkk** [comment]
Many thanks for the cleanup 👍

**Comment by psike01** [comment]
> Fix Aqara L1-350 ceiling light support. Current version has broken support of power on behaviour and color temperature restore at startup (report unsupported attributes).
> 
> `aqaraOpple` cluster is added to support additional attributes. After this, 'device temperature' and 'power outage count' seems to have the correct values. 'power_outage_memory' is also operated via `aqaraOpple` cluster.
> 
> The following data was observed from `aqaraOpple` cluster:
> 
> ```
> {
>     "3": 33,          <- device temperature
>     "5": 106,         <- outage count
>     "9": 256,         <- ?
>     "10": 0,          <- ?
>     "12": 10,         <- ?
>     "13": 28,         <- ?
>     "17": 1,          <- ?
>     "100": 0,         <- current state
>     "101": 254,       <- brightness
>     "102": 153,       <- color temperature
>     "103": 425986700, <- = 0x19640A8C 0x1964 - 6500, 0x0A8C - 2700, min and max color temp
>     "154": 0,         <- ?
> }
> ```
> 
> `{"1299":6500,"1300":2700}`
> 
> `{"238":0}`

Hi
After what update my lamp can't check OTA update, recive error "Device didn't respond to OTA request". Date firmware 20220331. Last firware, which can be updated is  20230905.
Can you check this?

**Pull request merged**